dali/python/nvidia/dali/pipeline.py

```diff
@@ -1323,6 +1323,7 @@ def create_pipeline(*args, **kwargs):
                     po = (pipe_outputs,)
                 pipe.set_outputs(*po)
             return pipe
+        create_pipeline.is_pipeline_def = True  # Add `is_pipeline_def` attribute to the function marked as `@pipeline_def`
```

Comment:
I suggest we should at least keep them private by adding a leading underscore. This will allow us to change it in the future and suggest that this is just implementation detail.

In the future we can add a function like `is_pipeline_def` that would check for the presence of this field.

BTW, what if someone makes it false by hand? :P